File: Data Structure/Linked List/Intersection of Two Linked Lists/getIntersectionNode.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def getIntersectionNode(self, headA, headB):
        """
        :type head1, head1: ListNode
        :rtype: ListNode
        """
        a={}
        while headA:
            a[headA]=headA
            headA=headA.next
        while headB:
            if headB in a:
                return headB
            headB=headB.next
              
        return None
    
     

File: Data Structure/Linked List/Intersection of Two Linked Lists/getIntersectionNode.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        set<ListNode *>myset;
     
        while(headA)
        {
            myset.insert(headA);
            headA=headA->next;
          
        }
        
        while(headB)
        {
            if(myset.find(headB)!=myset.end())
            {return headB;}
            else{headB=headB->next;}
        }
        return NULL;
    }    
};
